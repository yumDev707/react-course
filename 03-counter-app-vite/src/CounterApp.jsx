import PropTypes, { func } from 'prop-types';
import {useState} from 'react';

export const CounterApp = ({value}) => {
    const [counter, setCounter] = useState(value); //Hook. Desestructuración de lo que retorna la función, que inicializamos en 0.
    
    const handleAdd = (event, string) => { 
        // console.log(string, '+1');
        setCounter(counter + 1);
    }

    const handleLess = () => setCounter(counter - 1);
    const handleReset = () => setCounter(value);

    return (
        <>
            <h1>CounterApp</h1>
            {/* <h2>{value}</h2> */}
            <h2 data-testid="init-val">{counter}</h2>

            <button onClick={(event) => handleAdd(event, 'Hey!')}>
                Click me!
            </button>

            <div class="button-container">
                <button onClick={(event) => handleAdd(event, 'Hey!')}>+1</button>
                <button onClick={handleLess}>-1</button>
                <button aria-label="btn-reset" onClick={handleReset}>Reset</button>
            </div>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number
}