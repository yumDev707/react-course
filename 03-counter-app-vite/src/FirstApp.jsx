// import {Fragment} from 'react';
import PropTypes from 'prop-types';

const getGretting = () => {
    return 'Hey! I am a function!'
}

const newMsg = {
    // title: 'Prueba de variable en HTML.', 
    main: 'Jenn'
}

export const FirstApp = ({title, subtitle, name}) => {
    return (
        //Fragmento implícito, siempre necesario. De esta forma no necesitamos importarlo.
        <>
            <h1>{title}</h1>
            <div>My first app using React</div>
            {/* <p>Yay!</p> */}
            <p>{JSON.stringify(newMsg)}</p>
            <p>{getGretting()}</p>
            <p>{title}</p>
            <p>{subtitle}</p>
            <p>{name}</p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number
}

FirstApp.defaultProps = {
    subtitle: 3,
    name: 'Jeeeenn',
    // title: 'No hay título'
}