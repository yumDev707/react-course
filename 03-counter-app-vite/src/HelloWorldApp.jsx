
export const HelloWorldApp = () => {
    return (<h1>Hello World!</h1>);
}

// export const HelloWorldApp = () => <h1>Hello World!</h1>;

// function App() {
//     return (<h1>Hello World!</h1>);
// }

// export default App;