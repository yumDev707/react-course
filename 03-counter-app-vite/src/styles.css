html, body {
    background-color: rgb(79, 56, 82);
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 0.5rem;
}

button {
    width: 200px;
    height: 75px;
    font-size: 1.3rem;
    cursor: pointer;
    text-align: center;
    vertical-align: auto;
    margin: auto;
}

div.button-container {
    margin-top: 50px;
    button {
        width: 80px;
        height: 80px;
        font-size: 1rem;
    }
}