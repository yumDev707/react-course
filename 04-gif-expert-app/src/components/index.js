//Archivo de barril.

export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';