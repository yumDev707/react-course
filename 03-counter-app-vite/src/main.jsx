import React from 'react';
import ReactDom from 'react-dom/client';

import {HelloWorldApp} from './HelloWorldApp';
import {FirstApp} from './FirstApp';
import {CounterApp} from './CounterApp';
import './styles.css';

ReactDom.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <HelloWorldApp /> */}
        {/* De esta forma enviamos prop (properties). El valor subtitle envía núms enteros. */}
        {/* <FirstApp title="It is Yum!" subtitle={123}/> */}
        {/* <FirstApp /> */}
        <CounterApp value={100}/>
    </React.StrictMode>
);