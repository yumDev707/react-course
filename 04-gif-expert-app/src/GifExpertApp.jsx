//Funcional component.
import {useState} from 'react';
import {AddCategory, GifGrid} from './components/index';    //No serÃ­a necesario incluir index, porque por defecto ya busca esa ruta.

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Spy x Family']);  //Init.

    const onAddCategory = (newCategory) => {
        if(categories.includes(newCategory)) return;
        setCategories([newCategory, ...categories]);    //Hacemos una copia del array e insertamos el nuevo valor.
    };

    return(
        <>
            <h1>GifExpertApp</h1>

            <AddCategory 
                onNewCategory={onAddCategory}
            />

            {categories.map(category => (
                    <GifGrid key={category} category={category} />
                )
            )}
        </>
    )
}