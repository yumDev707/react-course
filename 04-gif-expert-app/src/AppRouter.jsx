import { Route, Routes, Navigate } from 'react-router-dom';
import { GifExpertApp } from './GifExpertApp';
import { AboutPage } from './components/About/AboutPage';

export const AppRouter = () => {
  return (
    <Routes>
      <Route path="/" element={<GifExpertApp />} />
      <Route path="/about" element={<AboutPage />} />
      {/* Redirecci√≥n por defecto si la ruta no existe */}
      <Route path="*" element={<Navigate to="/" />} />
    </Routes>
  );
};