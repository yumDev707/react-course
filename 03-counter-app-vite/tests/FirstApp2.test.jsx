const { render, screen } = require("@testing-library/react");
import { FirstApp } from "../src/FirstApp";


describe('Tests on FirstApp File', () => {
    const title = "Hey! I am Jenn."
    const subtitle = "I am a subtitle!";

    test('Should do match with snapshot', () => {
        const {container} = render(<FirstApp title={title}/>);
        expect(container).toMatchSnapshot();
    });

    test('Should show a message', () => {
        render(<FirstApp title={title}/>);
        // screen.debug();
        expect(screen.getAllByText(title)).toBeTruthy();
    })

    test('Should show an h1 title', () => {
        render(<FirstApp title={title}/>);
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
    })

    test('Should show the subtitle given by props', () => {
        render(<FirstApp title={title} subtitle={subtitle}/>);
        expect(screen.getAllByText(subtitle).length).toBe(1);
    })
})