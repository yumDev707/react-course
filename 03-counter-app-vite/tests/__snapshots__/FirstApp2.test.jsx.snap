// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on FirstApp File Should do match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hey! I am Jenn.
  </h1>
  <div>
    My first app using React
  </div>
  <p>
    {"main":"Jenn"}
  </p>
  <p>
    Hey! I am a function!
  </p>
  <p>
    Hey! I am Jenn.
  </p>
  <p>
    3
  </p>
  <p>
    Jeeeenn
  </p>
</div>
`;
